A3A/addons/core/functions/FactoryOverhaul/fn_productionGUIControls.sqf

```diff
@@ -0,0 +1,192 @@
+//#include "fn_productionHandler.sqf"
+#include "fn_productionUnlocks.sqf"
+
+#define IDX_FACTORY_GUI     6969
+#define IDC_FACTORY_LIST    69691
+#define IDC_FACTORY_WEPS    69692
+
+#define IDC_ITEM            69693
+#define IDC_QTY             69694
+
+#define IDC_STARTSTOP_BTN   69695
+#define IDC_EXIT_BUTTON     69696
+
+FO_fn_GUI_addFactoriesToListbox = {
+    // Clear the list box in case it already has items
+    systemChat "Initialising GUI controls...";
+
+    disableSerialization; 
+    lbClear IDC_FACTORY_LIST;
+    lbClear IDC_FACTORY_WEPS;
+    ctrlSetText [IDC_ITEM, ""];
+    ctrlSetText [IDC_QTY, ""];
+
+    private _ownedFactories = call FO_fn_GUI_createFactoryTownPairs;
+
+    // Calls func to check whether the factory is owned by the Rebels, and adds it to the listbox.
+    { 
+      _town = _x select 1;
+      lbAdd[IDC_FACTORY_LIST, _town];
+    } forEach _ownedFactories;
+    
+};
+
+FO_fn_GUI_createFactoryTownPairs = {
+    private _ownedFactories = []; // [_factory,townName];
+
+    // Calls func to check whether the factory is owned by the Rebels, and adds it to the listbox.
+    {
+      _town = [_x] call FO_fn_getNearestTown;
+      // Push this to the array of [_factory,townName].
+      _ownedFactories pushBack [_x,_town];
+    } forEach call FO_fn_getOwnedFactories;
+    
+    //systemChat str _ownedFactories;
+
+    _ownedFactories;
+
+};
+
+
+FO_fn_GUI_addWeaponsToListbox = {
+    {
+		private _itemName  = getText (configFile >> "CfgWeapons" >> _x select 0 >> "displayName");
+		lbAdd [IDC_FACTORY_WEPS, _itemName];
+	} forEach call FO_fn_getUnlockedItems;
+};
+
+
+FO_fn_GUI_lookupFactoryIDbyName = {
+    params ["_factoryTownPairs","_text"];
+
+    private _selectedFactoryFromTownname = "";
+    {
+        if ((_x select 1) == _text) then {
+            _selectedFactoryFromTownname = _x select 0;
+        };
+    } forEach _factoryTownPairs;
+//systemChat str _selectedFactoryFromTownname;
+
+_selectedFactoryFromTownname;
+};
+
+/////////////////////////////////////////// TODO
```

Comment:
If the function isn't used at all, you can comment it out in a more reasonable way like so:
```sqf
/* TODO
private _myFunc = {
    [the quick] brown (fox);
    jumps {
        over the lazy dog;
    };
};
*/
```

Looks better than leaving the entire function in but trying to "block it off" with one-line comments.

Reply:
[FIX REQUIRED]